### PR TITLE
Use the XML reporting API

### PR DESCRIPTION
**Pull request by mulby**
This is close enough to done that I thought it was worth putting up for review. Feel free to take a pass.

I'm still working on adding unit tests, specifically around the luigi task (I only have the happy path at the moment).

@jab5569 @brianhw

TODO list:
- [x] PEP8 and pylint cleanup
- [x] Lots of docstrings
- [x] Tests for reports that aren't done immediately
- [x] Tests for the task handling errors
- [x] Tests for the timeout behavior

### CODE REVIEW

**Comment by brianhw** [comment]
LGTM.  

One question:  do we want to add some information to our Transactions in general that Paypal provides and Cybersource doesn't (at present)?  In particular, the datetime instead of merely the date.  

Also, the Cybersource and transaction code is very USD-centric, with the assumption of two decimal places.  This seems to make different assumptions about representation, but do we have any idea what will happen if/when we get different currencies?

**Comment by mulby** [comment]
@brianhw I updated this branch with additional tests, docstrings and pylint/pep8 fixes

File: edx/analytics/tasks/reports/paypal.py [hunk 1]
```diff
@@ -18,183 +30,656 @@
...
+    Find the direct descendant of `node` with the tag `child_name`. If the no child is found with that tag, an error is
+    raised.
+
+    Args:
+        node: an ElementTree.Element object representing a node in an XML document that is expected to have a child
+            with the name specified by the `child_name` parameter.
+        child_name: a string containing the name of the tag that is expected to be a direct descendant of `node`.
 
+    Raises:
+        PaypalMalformedResponseException: If the document does not have the expected structure.
```
Comment (brianhw) [inline edx/analytics/tasks/reports/paypal.py]: returns an Error now, not an Exception.

File: edx/analytics/tasks/reports/paypal.py [hunk 2]
```diff
@@ -18,183 +30,656 @@
...
 
-    def initialize(self):
-        log.debug('Initializing paypalrestsdk')
-        paypalrestsdk.configure({
-            'mode': self.client_mode,
-            'client_id': self.client_id,
-            'client_secret': self.client_secret
-        })
+    Raises:
+        PaypalMalformedResponseException: If the document does not have the expected structure.
```
Comment (brianhw) [inline edx/analytics/tasks/reports/paypal.py]: returns an Error now, not an Exception.

**Comment by brianhw** [comment]
Two doc nits, otherwise LGTM. :+1:

File: edx/analytics/tasks/reports/paypal.py (hunk 1 above)
Comment (mulby) [inline edx/analytics/tasks/reports/paypal.py]: Done

File: edx/analytics/tasks/reports/paypal.py (hunk 2 above)
Comment (mulby) [inline edx/analytics/tasks/reports/paypal.py]: Done

**Comment by johnalbaker** [comment]
:+1:

**Pull request merged**